engine/script/src/script_sys.cpp

```diff
@@ -1297,6 +1297,45 @@ union SaveLoadBuffer
         return 1;
     }
 
+    //undocummented function for debugger
+
+    static int m_DebuggerLightweightHook;
+    static void Sys_DebuggerLightweightHook(lua_State *L, lua_Debug *ar)
+    {
+        int top = lua_gettop(L);
+        lua_getinfo(L, "S", ar);
+        lua_rawgeti(L, LUA_REGISTRYINDEX, m_DebuggerLightweightHook);
+        // 1:fn
+        lua_pushstring(L, ar->source);
+        // 2: str
+        lua_pushnumber(L, ar->lastlinedefined);
+        // 3: num
```

Comment:
This notation is a bit strange imo.
Normally, we write what the stack actually looks like with the ccorrect indices, so that you see how the operations affect the stack.
E.g. 
```
// -3: fn
// -2: str
// -1: num
```